android/src/main/java/playground/android/CoilFragment.kt

```diff
@@ -1,14 +1,22 @@
 package playground.android
 
 import android.os.Bundle
-import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.fragment.app.Fragment
+import coil.load
+import coil.transform.CircleCropTransformation
+import coil.transform.RoundedCornersTransformation
+import java.io.File
 
 
 class CoilFragment : Fragment() {
 
+    private var ivFile: ImageView? = null
+    private var ivUrl: ImageView? = null
+    private var ivDrawable: ImageView? = null
```

Comment:
@LouisCAD do you think those views should be nullable? I know nothing about Android

Reply:
If not nullable, we can initialize them later, but if we use lateinit, there might be possibility of views not being initialized, which would lead to crash of the application, so, i think it would handle it correctly.